src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -465,31 +467,40 @@ private async Task InitializeWorkersAsync()
                     server.Uri,
                     _loggerFactory); // TODO: DI (FACAVAL) Pass appropriate logger. Channel facory should likely be a service.
             };
-            var configFactory = new WorkerConfigFactory(ScriptSettingsManager.Instance.Configuration, _startupLogger);
+
+            _functionDispatcher = new FunctionDispatcher(EventManager, server, channelFactory, _workerConfigs);
+
+            _eventSubscriptions.Add(EventManager.OfType<WorkerProcessErrorEvent>()
+                .Subscribe(evt =>
+                {
+                    HandleHostError(evt.Exception);
+                }));
+        }
+
+        internal static IEnumerable<WorkerConfig> GetWorkerConfigs(string language, ILogger logger)
+        {
+            var configFactory = new WorkerConfigFactory(ScriptSettingsManager.Instance.Configuration, logger);
```

Comment:
Let's please avoid new use of `ScriptSettingsManager`. If we must use configuration, we should just add that dependency to the constructor, but ideally, this config factory wouldn't be needed and `WorkerConfig` is just passed in as an `IOptions<WorkerConfig>`. or a collection of them.

Reply:
Sure will do.